VS Code terminal integration issue after release v28.1.1

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

### What happened?

Hello there,

I'll report my problem here with as much initial detail as possible. Hopefully someone can help me with an easy workaround or help me find a way to address this issue correctly on my end.

I have a custom devcontainer me and my team use with remote containerized development in VS Code. This devcontainer is built off of a minimal RHEL UBI9 base image and integrates custom corporate configurations and settings into it. I've been extending this image with a set of dev tools that includes oh my posh. The terminal customizations include ble.sh (bash line editor) oh my posh, fzf, and many other niceties. After oh my posh version 28.1.1 we've found that VS Code terminal integration isn't working quite right. I think it has to do with this line item in the release notes.

bash: append hook to PROMPT_COMMAND instead of prepending ([81beaee](https://github.com/JanDeDobbeleer/oh-my-posh/commit/81beaee891c3e69bf017415719b0a29a00b3358b))

The error manifested when using GitHub copilot with agents that execute commands in VS Code's integrated terminal. The user seen the agent inject the command, the command ran and gave output, but the agent kept waiting on the output and spins forever. I found it works like you'd expect with version 28.1.0 but spins forever when the agent tries to run a command with version 28.1.1. When you examine the terminal details between the versions you can see the problem.

VS Code terminal details with my devcontainer and version 28.1.0:

```
bash

Process ID (PID): 1261

Command line: /bin/bash --init-file /vscode/vscode-server/bin/linux-x64/bf9252a2fb45be6893dd8870c0bf37e2e1766d61/out/vs/workbench/contrib/terminal/common/scripts/shellIntegration-bash.sh

The following extensions have contributed to this terminal's environment:

Git: Enables the following features: git auth provider
GitHub Copilot Chat: Enables use of copilot-debug and copilot commands in the terminal
Python
Shell integration: Rich

Shell type: bash
Current working directory: /workspaces/DynamicRoutingService
Seen sequences: P, E, C, A, B, D
Prompt type: oh-my-posh
Prompt input: |
```

VS Code terminal details with my devcontainer and version 28.1.1:

```bash

Process ID (PID): 1393

Command line: /bin/bash --init-file /vscode/vscode-server/bin/linux-x64/bf9252a2fb45be6893dd8870c0bf37e2e1766d61/out/vs/workbench/contrib/terminal/common/scripts/shellIntegration-bash.sh

The following extensions have contributed to this terminal's environment:

Git: Enables the following features: git auth provider
GitHub Copilot Chat: Enables use of copilot-debug and copilot commands in the terminal
Python
Shell integration: Rich

Shell type: bash
Current working directory: /workspaces/azure-workspace
Seen sequences: P, E, C, D
Prompt type: oh-my-posh
Prompt input: _omp_hook
```

The change that's effecting GitHub copilot would be the missing seen sequences I believe. Without A and B, it doesn't know when the command started and stopped and so copilot spins forever. At least I believe that's what's happening.

I can't give you the full base image that my env uses as it has corporate info in it, but I can show you the dev tooling parts I'm installing on top of my corporate minimal RHEL UBI 9 base image.

```Dockerfile
# this image is based on registry.redhat.io/ubi9/ubi-minimal:latest
FROM registry.nexusrepomgr.dtc.corp/abtech/python3-12:latest

ENV PATH="/app/.venv/bin/:$PATH" \
    PYTHONUNBUFFERED=1 \
    PYTHONDONTWRITEBYTECODE=1 \
    PIP_NO_CACHE_DIR=off \
    PIP_DISABLE_PIP_VERSION_CHECK=on \
    PIP_NO_WARN_SCRIPT_LOCATION=on \
    PIP_DEFAULT_TIMEOUT=100

RUN microdnf install -y git tar unzip tzdata make procps findutils nano bat fzf eza zoxide ripgrep git-delta vim neovim tmux fastfetch btop azure-cli s-nail bind-utils

ENV TZ=America/Chicago
RUN ln -snf /usr/share/zoneinfo/$TZ /etc/localtime && echo $TZ > /etc/timezone

RUN mkdir -p /root/.posh && \
    curl -fsSL https://github.com/JanDeDobbeleer/oh-my-posh/releases/download/v28.1.1/posh-linux-amd64 -o /root/.posh/oh-my-posh && \
    chmod +x /root/.posh/oh-my-posh && \
    mkdir -p /root/.posh-themes && \
    curl -fsSL https://github.com/JanDeDobbeleer/oh-my-posh/releases/download/v28.1.1/themes.zip -o /root/.posh/themes.zip && \
    unzip /root/.posh/themes.zip -d /root/.posh-themes && \
    rm /root/.posh/themes.zip

RUN git clone --depth 1 https://github.com/akinomyoga/ble.sh.git /root/.ble.sh && \
    make -C /root/.ble.sh

RUN /app/.venv/bin/python -m pip install tldr

RUN echo '# Oh My Posh configuration' >> /root/.bashrc && \
    echo 'export PATH=$PATH:/root/.posh:' >> /root/.bashrc && \
    echo 'eval "$(oh-my-posh init bash --config /root/.posh-themes/aliens.omp.json)"' >> /root/.bashrc && \
    echo '' >> /root/.bashrc && \
    echo '# Better command colors' >> /root/.bashrc && \
    echo 'export CLICOLOR=1' >> /root/.bashrc && \
    echo 'export LSCOLORS=ExFxBxDxCxegedabagacad' >> /root/.bashrc && \
    echo '' >> /root/.bashrc && \
    echo '# ble.sh: enhances Bash with autosuggestions and syntax highlighting' >> /root/.bashrc && \
    echo 'source $HOME/.ble.sh/out/ble.sh' >> /root/.bashrc && \
    echo 'bleopt history_share=1' >> /root/.bashrc && \
    echo 'export HISTFILE="$HOME/.local/share/blesh/ble_history"' >> /root/.bashrc && \
    echo '' >> /root/.bashrc && \
    echo '# alias cat to bat (a cat replacement with wings)' >> /root/.bashrc && \
    echo 'alias cat="bat"' >> /root/.bashrc && \
    echo '' >> /root/.bashrc && \
    echo '# alias ls to eza, with additional aliases' >> /root/.bashrc && \
    echo 'alias ls="eza --icons"' >> /root/.bashrc && \
    echo 'alias ll="eza -l --icons"' >> /root/.bashrc && \
    echo 'alias la="eza -la --icons"' >> /root/.bashrc && \
    echo '' >> /root/.bashrc && \
    echo '# Add .local/bin to PATH if not already there' >> /root/.bashrc && \
    echo 'if [[ ":$PATH:" != *":$HOME/.local/bin:"* ]]; then' >> /root/.bashrc && \
    echo '    export PATH="$HOME/.local/bin:$PATH"' >> /root/.bashrc && \
    echo 'fi' >> /root/.bashrc && \
    echo '' >> /root/.bashrc && \
    echo 'eval "$(zoxide init bash)"' >> /root/.bashrc && \
    echo 'alias cd="z"' >> /root/.bashrc && \
    echo '' >> /root/.bashrc && \
    echo '# Setting fzf environment variables...' >> /root/.bashrc && \
    echo 'eval "$(fzf --bash)"' >> /root/.bashrc && \
    echo 'export FZF_DEFAULT_OPTS="--layout=reverse --preview '\''bat --color=always {}'\''"' >> /root/.bashrc && \
    echo 'export FZF_CTRL_T_COMMAND="find . -type f -not -path '\''./.git/*'\''"' >> /root/.bashrc && \
    echo 'export FZF_CTRL_T_OPTS="--height 100% --preview '\''bat --color=always {}'\''"' >> /root/.bashrc && \
    echo 'export FZF_CTRL_R_OPTS="--height 100% --preview '\''echo {}'\'' --preview-window up:3:wrap"' >> /root/.bashrc
```

What would be the correct way to get VS Code integrated terminal to see all the sequences, including A & B with version 28.1.1? Is it something I'm doing weirdly in my dev container configuration? Has anyone else had this issue with GitHub copilot running commands when using oh my posh version 28.1.1?

If you need any other information from me on this let me know.

Thanks,
Codeuh

### Theme

aliens

### What OS are you seeing the problem on?

Linux

### Which shell are you using?

bash

### Log output

```Shell
Version: 28.1.1

Shell: shell (5.1.8(1)-release)

Prompt:

 root@40c50be27e4d  /workspaces/azure-workspace  master  app 3.12.11  

Segments:

ConsoleTitle(false)                       -   0 ms
Session(true)                             -   2 ms
Path(true)                                -   0 ms
Git(true)                                 -   1 ms
Python(true)                              -   3 ms

Run duration: 4.926883ms

Cache path: /root/.cache/oh-my-posh

Config path: /root/.posh-themes/aliens.omp.json

Logs:

[DEBUG] 15:39:49.858 debug.go:39 → logging enabled, raw mode: true
[DEBUG] 15:39:49.859 store.go:]:137 → (session) key not found: prompt_count_cache
[TRACE] 15:39:49.859 store.go:](session prompt_count_cache) - 32.626µs
[TRACE] 15:39:49.859 terminal.go:setPromptCount() - 40.857µs
[TRACE] 15:39:49.859 terminal.go:GOOS() - 157ns
[DEBUG] 15:39:49.859 terminal.go:setPwd:102 → /workspaces/azure-workspace
[TRACE] 15:39:49.859 terminal.go:setPwd() - 26.711µs
[TRACE] 15:39:49.859 terminal.go:Init() - 74.189µs
[TRACE] 15:39:49.859 init.go:SessionID() - 500ns
[DEBUG] 15:39:49.859 home.go:12 → /root
[TRACE] 15:39:49.859 path.go:Path() - 21.348µs
[DEBUG] 15:39:49.859 store.go:init:94 → (session) loading toggle_cache
[DEBUG] 15:39:49.859 store.go:init:94 → (session) loading prompt_count_cache
[DEBUG] 15:39:49.859 store.go:init:94 → (session) loading CONFIG
[DEBUG] 15:39:49.859 store.go:init:94 → (session) loading CONFIG_SOURCE
[TRACE] 15:39:49.859 store.go:init(session bash.6c7cfe74-9f0f-408e-aaf4-1a39268f90a7.omp.cache) - 174.387µs
[TRACE] 15:39:49.859 path.go:Path() - 69ns
[DEBUG] 15:39:49.859 store.go:init:94 → (device) loading DSC_CONFIGURATION
[DEBUG] 15:39:49.859 store.go:init:94 → (device) loading environment_platform
[DEBUG] 15:39:49.859 store.go:init:94 → (device) loading DSC_SHELL
[DEBUG] 15:39:49.859 store.go:init:94 → (device) loading is_wsl
[DEBUG] 15:39:49.859 store.go:init:94 → (device) loading INITVERSIONBASH
[TRACE] 15:39:49.859 store.go:init(device omp.cache) - 202.046µs
[DEBUG] 15:39:49.859 store.go:]:137 → (device) key not found: reload
[TRACE] 15:39:49.859 store.go:](device reload) - 6.539µs
[DEBUG] 15:39:49.859 store.go:]:150 → (session) found entry: CONFIG - /gH5fwMBAQZDb25maWcB/4AAAR8BB1BhbGV0dGUB/4IAAQtEZWJ1Z1Byb21wdAH/hAABA1ZhcgH/jgABCFBhbGV0dGVzAf+QAAEJVmFsaWRMaW5lAf+EAAEPU2Vjb25kYXJ5UHJvbXB0Af+EAAEPVHJhbnNpZW50UHJvbXB0Af+EAAEJRXJyb3JMaW5lAf+EAAEETWFwcwH/lAABB1VwZ3JhZGUB/5gAAQdFeHRlbmRzAQwAAQtBY2NlbnRDb2xvcgEMAAEUQ29uc29sZVRpdGxlVGVtcGxhdGUBDAABA1BXRAEMAAEGU291cmNlAQwAAQZGb3JtYXQBDAABElRlcm1pbmFsQmFja2dyb3VuZAEMAAEOVG9vbFRpcHNBY3Rpb24BDAABBkJsb2NrcwH/ngABBUN5Y2xlAf+iAAENSVRlcm1GZWF0dXJlcwH/pAABCFRvb2x0aXBzAf+cAAEHVmVyc2lvbgEEAAENTWlncmF0ZUdseXBocwECAAEFQXN5bmMBAgABEFNoZWxsSW50ZWdyYXRpb24BAgABCkZpbmFsU3BhY2UBAgABDVVwZ3JhZGVOb3RpY2UBAgABDlBhdGNoUHdzaEJsZWVkAQIAAQtBdXRvVXBncmFkZQECAAEXRW5hYmxlQ3Vyc29yUG9zaXRpb25pbmcBAgAAABf/gQQBAQdQYWxldHRlAf+CAAEMAQwAAP4CCP+DAwEBB1NlZ21lbnQB/4QAASEBClByb3BlcnRpZXMB/4YAAQVDYWNoZQH/iAABBUFsaWFzAQwAAQ5MZWFkaW5nRGlhbW9uZAEMAAEPVHJhaWxpbmdEaWFtb25kAQwAAQhUZW1wbGF0ZQEMAAEKRm9yZWdyb3VuZAEMAAEOVGVtcGxhdGVzTG9naWMBDAABD1Bvd2VybGluZVN5bWJvbAEMAAEKQmFja2dyb3VuZAEMAAEGRmlsbGVyAQwAAQRUeXBlAQwAAQVTdHlsZQEMAAEWTGVhZGluZ1Bvd2VybGluZVN5bWJvbAEMAAETRm9yZWdyb3VuZFRlbXBsYXRlcwH/igABBFRpcHMB/4wAARNCYWNrZ3JvdW5kVGVtcGxhdGVzAf+KAAEJVGVtcGxhdGVzAf+KAAEORXhjbHVkZUZvbGRlcnMB/4wAAQ5JbmNsdWRlRm9sZGVycwH/jAABBU5lZWRzAf+MAAEHVGltZW91dAEEAAEITWF4V2lkdGgBBAABCE1pbldpZHRoAQQAAQhEdXJhdGlvbgEEAAEKTmFtZUxlbmd0aAEEAAEFSW5kZXgBBAABC0ludGVyYWN0aXZlAQIAAQdFbmFibGVkAQIAAQdOZXdsaW5lAQIAAQ9JbnZlcnRQb3dlcmxpbmUBAgABBUZvcmNlAQIAAQdUb2dnbGVkAQIAAAAT/4UEAQEDTWFwAf+GAAEMARAAAC3/hwMBAQVDYWNoZQH/iAABAgEIRHVyYXRpb24BDAABCFN0cmF0ZWd5AQwAAAAS/4kCAQEETGlzdAH/igABDAAAFv+LAgEBCFtdc3RyaW5nAf+MAAEMAAAn/40EAQEXbWFwW3N0cmluZ11pbnRlcmZhY2Uge30B/44AAQwBEAAALf+PAwEBCFBhbGV0dGVzAf+QAAECAQRMaXN0Af+SAAEIVGVtcGxhdGUBDAAAACn/kQQBARhtYXBbc3RyaW5nXWNvbG9yLlBhbGV0dGUB/5IAAQwB/4IAAD//kwMBAQZDb25maWcB/5QAAQMBCFVzZXJOYW1lAf+WAAEISG9zdE5hbWUB/5YAAQlTaGVsbE5hbWUB/5YAAAAT/5UEAQEDTWFwAf+WAAEMAQwAAFz/lwMBAQZDb25maWcB/5gAAQYBBlNvdXJjZQEMAAEISW50ZXJ2YWwBDAABBkxhdGVzdAEMAAEEQXV0bwECAAENRGlzcGxheU5vdGljZQECAAEFRm9yY2UBAgAAAB7/nQIBAQ9bXSpjb25maWcuQmxvY2sB/54AAf+aAAD/qv+ZAwEC/5oAAQwBBFR5cGUBDAABCUFsaWdubWVudAEMAAEGRmlsbGVyAQwAAQhPdmVyZmxvdwEMAAEOTGVhZGluZ0RpYW1vbmQBDAABD1RyYWlsaW5nRGlhbW9uZAEMAAEIU2VnbWVudHMB/5wAAQhNYXhXaWR0aAEEAAEITWluV2lkdGgBBAABB05ld2xpbmUBAgABBUZvcmNlAQIAAQVJbmRleAEEAAAAIP+bAgEBEVtdKmNvbmZpZy5TZWdtZW50Af+cAAH/hAAAFP+hAgEBBUN5Y2xlAf+iAAH/oAAAKv+fAwEC/6AAAQIBCkJhY2tncm91bmQBDAABCkZvcmVncm91bmQBDAAAABv/owIBAQ1JVGVybUZlYXR1cmVzAf+kAAEMAAD+AdL/gAoBA2NkbgEEMTY4aAAFIi9yb290Ly5wb3NoLXRoZW1lcy9hbGllbnMub21wLmpzb24BBGpzb24DAQEGcHJvbXB0AQRsZWZ0BQQEA+6CtgED7oKwASEge3sgLlVzZXJOYW1lIH19QHt7IC5Ib3N0TmFtZSB9fSABByNmZmZmZmYDByM2MUFGRUYCB3Nlc3Npb24BB2RpYW1vbmQAAQEFc3R5bGUGc3RyaW5nDAYABGZ1bGwFDSB7eyAuUGF0aCB9fSABByNmZmZmZmYCA+6CsAEHI0M2NzhERAIEcGF0aAEJcG93ZXJsaW5lAAYNIHt7IC5IRUFEIH19IAEHIzE5MzU0OQID7oKwAQcjOTVmZmE0AgNnaXQBCXBvd2VybGluZQAEHjx0cmFuc3BhcmVudCxiYWNrZ3JvdW5kPu6CsDwvPgED7oK0AV4ge3sgaWYgLkVycm9yIH19e3sgLkVycm9yIH19e3sgZWxzZSB9fXt7IGlmIC5WZW52IH19e3sgLlZlbnYgfX0ge3sgZW5kIH19e3sgLkZ1bGwgfX17eyBlbmQgfX0gAQcjZmZmZmZmAwcjRkY2NDcxAgZweXRob24BB2RpYW1vbmQAAAQGBAEA
[TRACE] 15:39:49.859 store.go:](session CONFIG) - 28.472µs
[TRACE] 15:39:49.859 gob.go:Restore() - 86.117µs
[TRACE] 15:39:49.859 gob.go:Get() - 124.547µs
[TRACE] 15:39:49.859 terminal.go:Shell() - 95ns
[TRACE] 15:39:49.859 terminal.go:Flags() - 51ns
[DEBUG] 15:39:49.859 store.go:]:137 → (session) key not found: template_cache
[TRACE] 15:39:49.859 store.go:](session template_cache) - 4.468µs
[TRACE] 15:39:49.859 cache.go:restoreCache() - 7.527µs
[TRACE] 15:39:49.859 terminal_unix.go:Root() - 1.209µs
[TRACE] 15:39:49.859 terminal.go:Shell() - 75ns
[TRACE] 15:39:49.859 terminal.go:Flags() - 46ns
[TRACE] 15:39:49.859 terminal.go:StatusCodes() - 92ns
[DEBUG] 15:39:49.859 store.go:]:150 → (device) found entry: is_wsl - false
[TRACE] 15:39:49.859 store.go:](device is_wsl) - 3.208µs
[TRACE] 15:39:49.859 terminal_unix.go:IsWsl() - 5.676µs
[TRACE] 15:39:49.859 terminal.go:Flags() - 49ns
[TRACE] 15:39:49.859 terminal.go:Flags() - 30ns
[DEBUG] 15:39:49.859 home.go:12 → /root
[DEBUG] 15:39:49.859 store.go:]:150 → (device) found entry: is_wsl - false
[TRACE] 15:39:49.859 store.go:](device is_wsl) - 2.456µs
[TRACE] 15:39:49.859 terminal_unix.go:IsWsl() - 4.875µs
[TRACE] 15:39:49.859 terminal.go:Flags() - 30ns
[TRACE] 15:39:49.859 terminal.go:Flags() - 30ns
[TRACE] 15:39:49.859 terminal.go:GOOS() - 47ns
[DEBUG] 15:39:49.859 terminal.go:User:236 → root
[TRACE] 15:39:49.859 terminal.go:User() - 3.053µs
[DEBUG] 15:39:49.859 terminal.go:Host:253 → 40c50be27e4d
[TRACE] 15:39:49.859 terminal.go:Host() - 4.889µs
[TRACE] 15:39:49.859 terminal.go:GOOS() - 51ns
[DEBUG] 15:39:49.859 store.go:]:150 → (device) found entry: environment_platform - redhat
[TRACE] 15:39:49.859 store.go:](device environment_platform) - 3.707µs
[DEBUG] 15:39:49.859 terminal.go:Getenv:64 → 3
[TRACE] 15:39:49.859 terminal.go:Getenv(SHLVL) - 8.291µs
[DEBUG] 15:39:49.859 writer.go:Init:107 → terminal program: vscode
[DEBUG] 15:39:49.859 writer.go:Init:108 → terminal shell: shell
[DEBUG] 15:39:49.859 terminal.go:Getenv:64 → NO DATA
[TRACE] 15:39:49.859 terminal.go:Getenv(OMP_CACHE_DISABLED) - 2.605µs
[DEBUG] 15:39:49.859 store.go:]:137 → (device) key not found: accent_color
[TRACE] 15:39:49.859 store.go:](device accent_color) - 3.705µs
[TRACE] 15:39:49.859 colors.go:SetAccentColor() - 6.308µs
[TRACE] 15:39:49.859 terminal.go:Shell() - 46ns
[DEBUG] 15:39:49.859 terminal.go:Getenv:64 → 5.1.8(1)-release
[TRACE] 15:39:49.859 terminal.go:Getenv(POSH_SHELL_VERSION) - 2.348µs
[DEBUG] 15:39:49.859 debug.go:PrintDebug:24 → segment: Title
[TRACE] 15:39:49.859 terminal.go:Flags() - 62ns
[DEBUG] 15:39:49.859 terminal.go:Getenv:64 → NO DATA
[TRACE] 15:39:49.859 terminal.go:Getenv(POSH_CURSOR_LINE) - 2.1µs
[DEBUG] 15:39:49.859 terminal.go:Getenv:64 → NO DATA
[TRACE] 15:39:49.859 terminal.go:Getenv(POSH_CURSOR_COLUMN) - 1.437µs
[TRACE] 15:39:49.859 terminal.go:Flags() - 29ns
[TRACE] 15:39:49.859 terminal.go:Flags() - 34ns
[TRACE] 15:39:49.859 terminal.go:Flags() - 64ns
[TRACE] 15:39:49.859 terminal.go:Flags() - 440ns
[DEBUG] 15:39:49.859 home.go:12 → /root
[TRACE] 15:39:49.859 terminal.go:GOOS() - 65ns
[DEBUG] 15:39:49.859 home.go:12 → /root
[DEBUG] 15:39:49.859 segment.go:Execute:109 → segment: Python
[TRACE] 15:39:49.859 terminal.go:GOOS() - 91ns
[DEBUG] 15:39:49.859 segment.go:Execute:109 → segment: Path
[DEBUG] 15:39:49.859 store.go:]:150 → (session) found entry: toggle_cache - map[]
[TRACE] 15:39:49.859 store.go:](session toggle_cache) - 6.917µs
[DEBUG] 15:39:49.859 segment.go:isToggled:230 → no toggles found
[DEBUG] 15:39:49.859 store.go:]:150 → (session) found entry: toggle_cache - map[]
[TRACE] 15:39:49.859 store.go:](session toggle_cache) - 7.197µs
[DEBUG] 15:39:49.859 segment.go:isToggled:230 → no toggles found
[DEBUG] 15:39:49.859 properties.go:GetBool:40 → display_cygpath: false
[TRACE] 15:39:49.859 terminal.go:GOOS() - 52ns
[TRACE] 15:39:49.859 separator.go:Separator() - 50ns
[TRACE] 15:39:49.859 terminal.go:Shell() - 131ns
[TRACE] 15:39:49.859 terminal.go:Flags() - 359ns
[DEBUG] 15:39:49.860 properties.go:GetStringArray:70 → extensions: [*.py *.ipynb pyproject.toml venv.bak]
[DEBUG] 15:39:49.860 properties.go:GetStringArray:70 → folders: [.venv venv virtualenv venv-win pyenv-win]
[DEBUG] 15:39:49.860 properties.go:GetBool:40 → home_enabled: false
[DEBUG] 15:39:49.860 home.go:12 → /root
[DEBUG] 15:39:49.860 properties.go:GetBool:40 → fetch_virtual_env: true
[TRACE] 15:39:49.859 terminal.go:GOOS() - 27ns
[TRACE] 15:39:49.860 terminal.go:Flags() - 339ns
[DEBUG] 15:39:49.860 properties.go:GetBool:40 → mapped_locations_enabled: true
[DEBUG] 15:39:49.860 properties.go:GetString:46 → windows_registry_icon: 
[TRACE] 15:39:49.860 terminal.go:GOOS() - 81ns
[DEBUG] 15:39:49.860 properties.go:GetString:46 → windows_registry_icon: 
[DEBUG] 15:39:49.860 segment.go:Execute:109 → segment: Git
[DEBUG] 15:39:49.860 home.go:12 → /root
[TRACE] 15:39:49.860 separator.go:Separator() - 57ns
[DEBUG] 15:39:49.860 store.go:]:150 → (session) found entry: toggle_cache - map[]
[DEBUG] 15:39:49.860 terminal.go:CommandPath:304 → /app/.venv/bin/python
[TRACE] 15:39:49.860 terminal.go:CommandPath(python) - 30.906µs
[TRACE] 15:39:49.860 store.go:](session toggle_cache) - 10.005µs
[DEBUG] 15:39:49.860 home.go:12 → /root
[DEBUG] 15:39:49.860 segment.go:isToggled:230 → no toggles found
[TRACE] 15:39:49.860 terminal.go:GOOS() - 78ns
[DEBUG] 15:39:49.860 segment.go:Execute:109 → segment: Session
[DEBUG] 15:39:49.860 properties.go:GetBool:40 → disable_with_jj: false
[DEBUG] 15:39:49.860 store.go:]:150 → (session) found entry: toggle_cache - map[]
[TRACE] 15:39:49.860 store.go:](session toggle_cache) - 10.517µs
[DEBUG] 15:39:49.860 segment.go:isToggled:230 → no toggles found
[TRACE] 15:39:49.860 terminal.go:GOOS() - 62ns
[TRACE] 15:39:49.860 terminal.go:GOOS() - 60ns
[DEBUG] 15:39:49.860 properties.go:GetString:46 → home_icon: ~
[DEBUG] 15:39:49.860 properties.go:GetKeyValueMap:64 → mapped_locations: map[]
[DEBUG] 15:39:49.860 terminal.go:Getenv:64 → NO DATA
[TRACE] 15:39:49.860 separator.go:Separator() - 47ns
[TRACE] 15:39:49.860 terminal.go:Getenv(SSH_CONNECTION) - 7.73µs
[TRACE] 15:39:49.860 terminal.go:GOOS() - 28ns
[DEBUG] 15:39:49.860 terminal.go:Getenv:64 → NO DATA
[TRACE] 15:39:49.860 terminal.go:GOOS() - 28ns
[TRACE] 15:39:49.860 terminal.go:Getenv(SSH_CLIENT) - 1.937µs
[TRACE] 15:39:49.860 terminal.go:ResolveSymlink(/workspaces/azure-workspace) - 18.245µs
[TRACE] 15:39:49.860 separator.go:Separator() - 28ns
[DEBUG] 15:39:49.860 store.go:]:150 → (device) found entry: environment_platform - redhat
[TRACE] 15:39:49.860 terminal.go:GOOS() - 27ns
[TRACE] 15:39:49.860 terminal.go:GOOS() - 28ns
[TRACE] 15:39:49.860 terminal.go:GOOS() - 28ns
[TRACE] 15:39:49.860 terminal.go:GOOS() - 53ns
[TRACE] 15:39:49.860 terminal.go:GOOS() - 30ns
[DEBUG] 15:39:49.860 terminal.go:HasFilesInDir:151 → false
[TRACE] 15:39:49.860 terminal.go:HasFilesInDir(pyvenv.cfg) - 92.701µs
[DEBUG] 15:39:49.860 properties.go:GetString:46 → gitdir_format: 
[DEBUG] 15:39:49.860 properties.go:GetString:46 → style: full
[DEBUG] 15:39:49.860 properties.go:GetStringArray:70 → cycle: []
[DEBUG] 15:39:49.860 properties.go:GetString:46 → folder_separator_template: 
[DEBUG] 15:39:49.860 properties.go:GetString:46 → folder_separator_icon: /
[DEBUG] 15:39:49.860 properties.go:GetBool:40 → cycle_folder_separator: false
[DEBUG] 15:39:49.860 properties.go:GetString:46 → folder_format: %s
[DEBUG] 15:39:49.860 properties.go:GetString:46 → edge_format: %s
[DEBUG] 15:39:49.860 properties.go:GetString:46 → left_format: %s
[DEBUG] 15:39:49.860 properties.go:GetString:46 → right_format: %s
[TRACE] 15:39:49.860 terminal.go:Flags() - 49ns
[TRACE] 15:39:49.860 terminal.go:GOOS() - 47ns
[TRACE] 15:39:49.860 terminal.go:StackCount() - 73ns
[DEBUG] 15:39:49.860 terminal.go:FileContent:212 ↓
    home = /usr/bin
    include-system-site-packages = false
    version = 3.12.11
    executable = /usr/bin/python3.12
    command = /usr/bin/python3.12 -m venv /app/.venv
    
[TRACE] 15:39:49.860 terminal.go:FileContent(/app/.venv/pyvenv.cfg) - 43.134µs
[TRACE] 15:39:49.860 terminal_unix.go:DirIsWritable(/workspaces/azure-workspace) - 10.663µs
[DEBUG] 15:39:49.860 properties.go:GetBool:40 → folder_name_fallback: true
[DEBUG] 15:39:49.860 properties.go:GetStringArray:70 → default_venv_names: [.venv venv]
[DEBUG] 15:39:49.860 terminal.go:Getenv:64 → /app/.venv
[TRACE] 15:39:49.860 terminal.go:Getenv(VIRTUAL_ENV) - 2.667µs
[DEBUG] 15:39:49.860 python.go:loadContext:93 → virtual env name: .venv
[DEBUG] 15:39:49.860 python.go:loadContext:97 → virtual env name (fallback): app
[DEBUG] 15:39:49.860 properties.go:GetBool:40 → display_default: true
[DEBUG] 15:39:49.860 properties.go:GetBool:40 → fetch_version: true
[DEBUG] 15:39:49.860 store.go:]:137 → (device) key not found: version_python
[TRACE] 15:39:49.860 store.go:](device version_python) - 5.259µs
[DEBUG] 15:39:49.860 terminal.go:CommandPath:297 → /app/.venv/bin/python
[TRACE] 15:39:49.860 terminal.go:CommandPath(python) - 2.962µs
[DEBUG] 15:39:49.860 terminal.go:Getenv:64 → NO DATA
[TRACE] 15:39:49.860 terminal.go:Getenv(PYENV_ROOT) - 2.105µs
[ERROR] 15:39:49.860 language.go:setVersion:212 → executable at /app/.venv/bin/python is not a pyenv shim
[DEBUG] 15:39:49.860 terminal.go:CommandPath:297 → /app/.venv/bin/python
[TRACE] 15:39:49.860 terminal.go:CommandPath(python) - 2.243µs
[TRACE] 15:39:49.860 terminal.go:HasCommand(python) - 4.398µs
[TRACE] 15:39:49.860 terminal.go:HasParentFilePath(.git) - 622.776µs
[DEBUG] 15:39:49.860 properties.go:GetBool:40 → fetch_bare_info: false
[DEBUG] 15:39:49.860 store.go:]:150 → (device) found entry: is_wsl - false
[TRACE] 15:39:49.860 store.go:](device is_wsl) - 6.068µs
[TRACE] 15:39:49.860 terminal_unix.go:IsWsl() - 9.599µs
[TRACE] 15:39:49.860 terminal_unix.go:IsWsl2() - 12.147µs
[TRACE] 15:39:49.860 terminal.go:GOOS() - 45ns
[DEBUG] 15:39:49.860 terminal.go:CommandPath:304 → /usr/bin/git
[TRACE] 15:39:49.860 terminal.go:CommandPath(git) - 64.106µs
[TRACE] 15:39:49.860 terminal.go:HasCommand(git) - 66.328µs
[DEBUG] 15:39:49.860 home.go:12 → /root
[TRACE] 15:39:49.860 terminal.go:GOOS() - 42ns
[TRACE] 15:39:49.860 terminal.go:GOOS() - 28ns
[DEBUG] 15:39:49.860 properties.go:GetBool:40 → fetch_user: false
[DEBUG] 15:39:49.860 properties.go:GetString:46 → source: cli
[DEBUG] 15:39:49.860 properties.go:GetBool:40 → fetch_status: false
[DEBUG] 15:39:49.861 terminal.go:FileContent:212 ↓
    ref: refs/heads/master
    
[TRACE] 15:39:49.861 terminal.go:FileContent(/workspaces/azure-workspace/.git/HEAD) - 778.048µs
[DEBUG] 15:39:49.861 git.go:updateHEADReference:974 → HEADRef: ref: refs/heads/master
[DEBUG] 15:39:49.861 git.go:updateHEADReference:999 → current HEAD is a branch: master
[DEBUG] 15:39:49.861 properties.go:GetString:46 → branch_icon: 
[DEBUG] 15:39:49.861 properties.go:GetKeyValueMap:64 → mapped_branches: map[]
[DEBUG] 15:39:49.861 properties.go:GetString:46 → branch_template: 
[DEBUG] 15:39:49.861 properties.go:GetBool:40 → fetch_upstream_icon: false
[DEBUG] 15:39:49.862 terminal.go:RunCommand:280 → NO DATA
[TRACE] 15:39:49.862 terminal.go:RunCommand(who am i) - 2.239182ms
[DEBUG] 15:39:49.862 text.go:patchTemplate:149 →  {{ .UserName }}@{{ .HostName }} 
[DEBUG] 15:39:49.862 terminal.go:RunCommand:280 → Python 3.12.11
[TRACE] 15:39:49.862 terminal.go:RunCommand(python --version) - 2.287304ms
[DEBUG] 15:39:49.862 properties.go:GetString:46 → version_url_template: https://docs.python.org/release/{{ .Major }}.{{ .Minor }}.{{ .Patch }}/whatsnew/changelog.html#python-{{ .Major }}-{{ .Minor }}-{{ .Patch }}
[DEBUG] 15:39:49.862 text.go:patchTemplate:149 → https://docs.python.org/release/{{ .Data.Major }}.{{ .Data.Minor }}.{{ .Data.Patch }}/whatsnew/changelog.html#python-{{ .Data.Major }}-{{ .Data.Minor }}-{{ .Data.Patch }}
[DEBUG] 15:39:49.863 text.go:patchTemplate:149 →  {{ .Data.Path }} 
[DEBUG] 15:39:49.863 properties.go:GetString:46 → cache_duration: none
[TRACE] 15:39:49.863 store.go:](device version_python) - 2.429µs
[DEBUG] 15:39:49.863 text.go:patchTemplate:149 →  {{ .Data.HEAD }} 
[DEBUG] 15:39:49.863 text.go:patchTemplate:149 →  {{ if .Data.Error }}{{ .Data.Error }}{{ else }}{{ if .Data.Venv }}{{ .Data.Venv }} {{ end }}{{ .Data.Full }}{{ end }} 
[TRACE] 15:39:49.863 terminal.go:Shell() - 199ns
[TRACE] 15:39:49.863 terminal.go:Shell() - 50ns
[TRACE] 15:39:49.863 path.go:Path() - 141ns
```

--- a/src/shell/scripts/omp.bash
+++ b/src/shell/scripts/omp.bash
@@ -158,7 +158,13 @@ function _omp_install_hook() {
         prompt_command+=("$cmd")
     done
 
-    PROMPT_COMMAND=("${prompt_command[@]}" _omp_hook)
+    # When VS Code shell integration is active, prepend so _omp_hook sets PS1 before
+    # VS Code's __vsc_prompt_cmd_original wraps it with A/B shell integration sequences.
+    if [[ "${TERM_PROGRAM-}" == "vscode" ]]; then
+        PROMPT_COMMAND=(_omp_hook "${prompt_command[@]}")
+    else
+        PROMPT_COMMAND=("${prompt_command[@]}" _omp_hook)
+    fi
 }
 
 _omp_install_hook

